src/actions/route_helpers.cr

```diff
@@ -0,0 +1,61 @@
+module DppmRestApi::Actions
+  # A series of utility functions that are useful to all of the Action blocks
+  module RouteHelpers
+    macro fmt_route(route = "", namespace = false)
+    {{ "/" + @type.stringify
+         .downcase
+         .gsub(/^DppmRestApi::Actions::/, "")
+         .gsub(/::/, "/") }} {% if namespace %} + "/:namespace" {% end %} + ( {{route}} || "")
+    end
+
+    # the "last" or -1st block argument is selected because context is always the
+    # argument -- either the only or the second of two
+    {% for method in %w(get post put delete ws options) %}
+    macro relative_{{method.id}}(route, &block)
+      {{method.id}} fmt_route(\{{route}}) do |\{{block.args.splat}}|
+        namespace = \{{block.args[-1]}}.params.query["namespace"]? || DPPM::Prefix.default_group
+        \{{block.body}}
+      end
+    end
+    {% end %}
+
+    # This method parses a boolean value from a query parameter. It returns
+    # true when the query parameter contains any string *except* for `"false"`
+    # or `"0"`.
+    def parse_boolean_param(key : String, from context : HTTP::Server::Context) : Bool
+      if param = context.params.query[key]?
+        return true unless {"false", "0"}.includes? param.downcase
```

Comment:
Any reason I miss to support also `0`, in addition to `false`?

Reply:
Idk it just seems common to assume 1 and 0 will work for boolean operations.

Reply:
Not really, we use `true` and `false` everywhere else (json, con, CLI, Crystal).
It seems also more common for other APIs to pass `true` and `false` (which are strings)

Reply:
The `param.downcase` means  that `FALSE` is also supported?!

Reply:
I propose to only support `true` and `false`, else raise.

Reply:
I hadn't thought of raising `BadRequest`. Counter-proposal -- only support the existence or lack thereof and raise if there is *any* text given in a boolean parameter. `?libraries&something` would be the only option, anything else ( `?libaries=yes&something=false`) would raise `BadRequest`.

Reply:
Wait, why having this possibility?
I think I will rename `query` to `info`, because that's the info present in `pkg.con` file.
Querying the configuration is another endpoint, `/config`, and abstract the origin of the configuration.

Reply:
Is this comment about a different section of code? I'm confused